### PR TITLE
feat(apps/legacy-api): Legacy API - stats

### PR DESCRIPTION
**Pull request by eli-lim**
What this PR does / why we need it:
Implementation of legacy-api to remap the legacy stats API (https://api.defichain.com/v1/stats).

Which issue(s) does this PR fixes?:
Fixes part of https://github.com/DeFiCh/jellyfish/issues/1024

### CODE REVIEW

**Comment by fuxingloh** [comment]
Btw conflict

**Comment by fuxingloh** [comment]
Btw conflict

**Review by joeldavidw — APPROVED** [review]

File: apps/legacy-api/src/controllers/stats/LegacyStatsProvider.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+@Injectable()
+export class TestnetLegacyStatsProvider extends MainnetLegacyStatsProvider {
+  constructor (clientProvider: WhaleApiClientProvider) {
+    super(clientProvider)
+    this.api = clientProvider.getClient('testnet')
+    this.chain = 'test'
+  }
+
+  calculateBlockSubsidy (blockHeight: number): number {
+    const blockSubsidy = new BlockSubsidy(TestNetCoinbaseSubsidyOptions)
```
Comment (fuxingloh) [inline apps/legacy-api/src/controllers/stats/LegacyStatsProvider.ts:186]: Should be a class variable, as it's computational heavy Object.

File: apps/legacy-api/src/controllers/stats/LegacyStatsProvider.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+  async getListCommunities (): Promise<LegacyListCommunities> {
+    const communityBalances = await this.api.rpc.call<CommunityBalanceData>('listcommunitybalances', [], 'number')
+    return {
+      AnchorReward: new BigNumber(communityBalances.AnchorReward).toNumber(),
+      Burnt: new BigNumber(communityBalances.Burnt).toFixed(DECIMAL_PLACES)
+    }
+  }
+
+  calculateBlockSubsidy (blockHeight: number): number {
+    const blockSubsidy = new BlockSubsidy(MainNetCoinbaseSubsidyOptions)
```
Comment (fuxingloh) [inline apps/legacy-api/src/controllers/stats/LegacyStatsProvider.ts:155]: Same for this.

> Should be a class variable, as it's computational heavy Object.

**Review by fuxingloh — APPROVED** [review]
Suggestions for improvement, for future PR.

**Pull request merged**